From 3b4e41cb7c3a85778545809ec68ebbd0c8c1700b Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Wed, 20 Jan 2021 19:05:08 +0200
Subject: [PATCH] [SW-29389] gaudi: security: protect psoc etr

The ETR (Embedded Trace Router) is capable of stopping ARM and causing
other security risks.
Therefore it must be protected.
It does not have the Protection Bits, so we are forced to use the Range
Register.

Change-Id: I37f89803d10c70f37d12e239604ddc386bf88876
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/security.c b/gaudi/security.c
index e0c47d0..874bb63 100644
--- a/gaudi/security.c
+++ b/gaudi/security.c
@@ -1232,6 +1232,11 @@
 	min = mmNIC3_TS_BASE - 1;
 	max = mmNIC4_MAC_CH0_BASE;
 	preboot_secure_lbw_rr(14, min, max);
+
+	/* PSOC_ETR up to PSOC_ETF: RR_15 */
+	min = mmPSOC_ETR_BASE - 1;
+	max = mmPSOC_ETF_BASE;
+	preboot_secure_lbw_rr(15, min, max);
 }
 
 static void preboot_secure_hbw_rr(int rr, u64 base, u64 mask)
